### PR TITLE
Cleanup various deprecated code and options

### PR DESCRIPTION
**Pull request by ste93cry**
| Q             | A
| ------------- | ---
| Branch?       | 3.0
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | yes
| Deprecations? | no
| License       | MIT

Mixed cleanup of deprecated methods, constants and options. Remaining deprecations will be addressed when refactoring the code and thus will be done in separate PRs

### CODE REVIEW

File: UPGRADE-3.0.md [hunk 1]
```diff
@@ -15,3 +15,10 @@
 - The signature of the `Event::getId()` method changed to return an instance of the `EventId` class instead of a `string`
 - The signature of the `Options::getDsn()` method changed to always return an instance of the `Dsn` class instead of a `string`
 - Removed the `Options::getProjectId`, `Options::getPublicKey` and `Options::getSecretKey` methods, use `Options::getDsn()` instead.
+- Removed the `Breadcrumb::LEVEL_CRITICAL` constant. Use `Breadcrumb::LEVEL_FATAL` instead
+- Removed the `Breadcrumb::levelFromErrorException()` method
+- Removedc the `PluggableHttpClientFactory` class
```
Comment (Jean85) [inline UPGRADE-3.0.md:20]:
```suggestion
- Removed the `PluggableHttpClientFactory` class
```

File: UPGRADE-3.0.md [hunk 2]
```diff
@@ -15,3 +15,10 @@
...
+- Removed the `ClientBuilderInterface::setUriFactory()`, `ClientBuilderInterface::setMessageFactory()`, `ClientBuilderInterface::setTransport()`, `ClientBuilderInterface::setHttpClient()`, `ClientBuilderInterface::addHttpClientPlugin()` and `ClientBuilderInterface::removeHttpClientPlugin()` methods. Use `ClientBuilderInterface::setTransportFactory()` instead
```
Comment (Jean85) [inline UPGRADE-3.0.md:21]: Can we rephrase this (and the others) with multiple methods like this? It's should be easier to scan:
```suggestion
- Removed the following methods from `ClientBuilderInterface`, use `ClientBuilderInterface::setTransportFactory()` instead: 
   - `ClientBuilderInterface::setUriFactory()`
   - `ClientBuilderInterface::setMessageFactory()`
   - `ClientBuilderInterface::setTransport()`
   - `ClientBuilderInterface::setHttpClient()`
   - `ClientBuilderInterface::addHttpClientPlugin()`
   - `ClientBuilderInterface::removeHttpClientPlugin()`
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,12 @@
 - [BC BREAK] Change the default value for the `error_types` option from `E_ALL` to the value get from `error_reporting()` (#1037)
 - [BC BREAK] Remove deprecated code to return the event ID as a `string` rather than an object instance from the transport, the client and the hub (#1036)
 - [BC BREAK] Remove some deprecated methods from the `Options` class.
+- [BC BREAK] Remove the deprecated code from the `ModulesIntegration` integration (#1045)
+- [BC BREAK] Remove the deprecated code from the `RequestIntegration` integration (#1045)
+- [BC BREAK] Remove the deprecated code from the `Breadcrumb` class (#1045)
```
Comment (stayallive) [inline CHANGELOG.md:12]: I think you meant this PR for these changes?

```suggestion
- [BC BREAK] Remove the deprecated code from the `ModulesIntegration` integration (#1047)
- [BC BREAK] Remove the deprecated code from the `RequestIntegration` integration (#1047)
- [BC BREAK] Remove the deprecated code from the `Breadcrumb` class (#1047)
```

File: UPGRADE-3.0.md [hunk 4]
```diff
@@ -15,3 +15,21 @@
...
+  - `addHttpClientPlugin()`
+  - `removeHttpClientPlugin()`.
+- Removed the following methods from the `Options` class, use the `IgnoreErrorsIntegration` integration instead:
+  - `getExcludedExceptions()`
+  - `Options::setExcludedExceptions()`
+  - `Options::isExcludedException()`
+  - `Options::getProjectRoot()`
+  - `Options::setProjectRoot()`
+- Removed the `Context::CONTEXT_USER`, `Context::CONTEXT_RUNTIME`, `Context::CONTEXT_TAGS`, `Context::CONTEXT_EXTRA`, `Context::CONTEXT_SERVER_OS` constants
+- The signature of the `Scope::setUser()` method changed to not accept anymore the `$merge` parameter
```
Comment (stayallive) [inline UPGRADE-3.0.md:35]:
```suggestion
- The signature of the `Scope::setUser()` method changed to not accept the `$merge` parameter anymore
```

File: UPGRADE-3.0.md [hunk 5]
```diff
@@ -15,3 +15,21 @@
 - The signature of the `Event::getId()` method changed to return an instance of the `EventId` class instead of a `string`
 - The signature of the `Options::getDsn()` method changed to always return an instance of the `Dsn` class instead of a `string`
 - Removed the `Options::getProjectId`, `Options::getPublicKey` and `Options::getSecretKey` methods, use `Options::getDsn()` instead.
+- Removed the `Breadcrumb::LEVEL_CRITICAL` constant. Use `Breadcrumb::LEVEL_FATAL` instead
```
Comment (stayallive) [inline UPGRADE-3.0.md:18]:
```suggestion
- Removed the `Breadcrumb::LEVEL_CRITICAL` constant, use `Breadcrumb::LEVEL_FATAL` instead
```

File: UPGRADE-3.0.md [hunk 6]
```diff
@@ -15,3 +15,21 @@
 - The signature of the `Event::getId()` method changed to return an instance of the `EventId` class instead of a `string`
 - The signature of the `Options::getDsn()` method changed to always return an instance of the `Dsn` class instead of a `string`
 - Removed the `Options::getProjectId`, `Options::getPublicKey` and `Options::getSecretKey` methods, use `Options::getDsn()` instead.
```
Comment (stayallive) [inline UPGRADE-3.0.md:17]:
```suggestion
- Removed the `Options::getProjectId`, `Options::getPublicKey` and `Options::getSecretKey` methods, use `Options::getDsn()` instead
```

**Pull request merged**